packages/ckeditor5-core/src/editor/utils/elementapimixin.ts

```diff
@@ -44,15 +44,17 @@ export default function ElementApiMixin<Base extends Constructor<Editor>>( base:
 
 			// The data returned by the editor might be unsafe, so we want to prevent rendering
 			// unsafe content inside the source element different than <textarea>, which is considered
-			// secure. This behaviour could be changed by setting the `updateSourceElementOnDestroy`
+			// secure. This behavior could be changed by setting the `updateSourceElementOnDestroy`
 			// configuration option to `true`.
 			if ( !shouldUpdateSourceElement && !isSourceElementTextArea ) {
 				setDataInElement( this.sourceElement, '' );
 
 				return;
 			}
 
-			setDataInElement( this.sourceElement, data );
+			const dataToSet = data || this.data.get();
```

Comment:
What if `data` is empty string and `this.data.get()` is not?

And please add test coverage for that case.